debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -388,45 +389,53 @@ protected void execute() {
+                    allTableIds.addAll(tableIds);
+                    allTableIds.stream()
+                               .filter(id -> isRunning()) // ignore all subsequent tables if this reader is stopped
+                               .forEach(tableId -> schema.applyDdl(source, tableId.schema(),
+                                                                   "DROP TABLE IF EXISTS " + quote(tableId),
+                                                                   this::enqueueSchemaChanges));
+
+                    // Add a DROP DATABASE statement for each database that we no longer know about ...
+                    schema.tables().tableIds().stream().map(TableId::catalog)
+                          .filter(Predicates.not(readableDatabaseNames::contains))
+                          .filter(id -> isRunning()) // ignore all subsequent tables if this reader is stopped
+                          .forEach(missingDbName -> schema.applyDdl(source, missingDbName,
+                                                                    "DROP DATABASE IF EXISTS " + quote(missingDbName),
+                                                                    this::enqueueSchemaChanges));
+
+                    // Now process all of our tables for each database ...
+                    for (Map.Entry<String, List<TableId>> entry : tableIdsByDbName.entrySet()) {
                         if (!isRunning()) break;
-                        sql.set("SHOW CREATE TABLE " + quote(tableId));
-                        mysql.query(sql.get(), rs -> {
-                            if (rs.next()) {
-                                schema.applyDdl(source, dbName, rs.getString(2), this::enqueueSchemaChanges);
-                            }
-                        });
+                        String dbName = entry.getKey();
+                        // First drop, create, and then use the named database ...
+                        schema.applyDdl(source, dbName, "DROP DATABASE IF EXISTS " + quote(dbName), this::enqueueSchemaChanges);
+                        schema.applyDdl(source, dbName, "CREATE DATABASE " + quote(dbName), this::enqueueSchemaChanges);
+                        schema.applyDdl(source, dbName, "USE " + quote(dbName), this::enqueueSchemaChanges);
+                        for (TableId tableId : entry.getValue()) {
+                            if (!isRunning()) break;
+                            sql.set("SHOW CREATE TABLE " + quote(tableId));
+                            mysql.query(sql.get(), rs -> {
+                                if (rs.next()) {
+                                    schema.applyDdl(source, dbName, rs.getString(2), this::enqueueSchemaChanges);
+                                }
+                            });
+                        }
                     }
+                    context.makeRecord().regenerate();
+                }
+                // most likely, something went wrong while writing the history topic
```

Comment:
I think we are using different formatting for `catch` clauses

Reply:
Yes, you are right. Let's talk about that one in the chat, I'll ping you.